hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/security/TestDelegationToken.java

```diff
@@ -376,4 +382,61 @@ public void testDelegationTokenIdentifierToString() throws Exception {
         " for SomeUser with renewer JobTracker",
         dtId.toStringStable());
   }
+
+  public static class MyDelegationTokenSecretManager extends
+      AbstractDelegationTokenSecretManager<DelegationTokenIdentifier> {
+    /**
+     * Create a secret manager
+     *
+     * @param delegationKeyUpdateInterval        the number of milliseconds for rolling
+     *                                           new secret keys.
+     * @param delegationTokenMaxLifetime         the maximum lifetime of the delegation
+     *                                           tokens in milliseconds
+     * @param delegationTokenRenewInterval       how often the tokens must be renewed
+     *                                           in milliseconds
+     * @param delegationTokenRemoverScanInterval how often the tokens are scanned
+     *                                           for expired tokens in milliseconds
+     */
+    public MyDelegationTokenSecretManager(long delegationKeyUpdateInterval,
+        long delegationTokenMaxLifetime, long delegationTokenRenewInterval,
+        long delegationTokenRemoverScanInterval) {
+      super(delegationKeyUpdateInterval,
```

Comment:
Doesn't this happen automatically?

Reply:
Yes, the ExpiredTokenRemover thread will operate automatically, which is selected to be called automatically in the test case to determine if an exception has been generated. If an exception has been generated, it indicates that it has not been fixed.Any suggestions for improvement?

Reply:
I mean the constructor doesnt really need to be declared explicitly.

Reply:
> I mean the constructor doesnt really need to be declared explicitly.
Thank you for your suggestion. Let me revise it.